inspirehep/modules/records/tasks.py

```diff
@@ -204,3 +210,38 @@ def actions():
         'success': success,
         'failures': [failure for failure in failures or []],
     }
+
+
+@shared_task(ignore_result=False, bind=True)
+def index_new_cited_records(self, pid_type, pid_value, db_version):
+    record = get_db_record(pid_type, pid_value)
+
+    if record.model.version_id < db_version:
+        backoff = 2 ** (self.request.retries + 1)
+        eta = datetime.datetime.now() + datetime.timedelta(seconds=backoff)
+        logger.warn(
+            'Record {} not yet updated to version {} on DB'.format((pid_type, pid_value), db_version)
+        )
+        raise self.retry(eta=eta)
```

Comment:
Here you can retry forever. Do you want to put a limit on the number of retries?

Reply:
yes, how many retries in your opinion? 
With 3 it would wait at least 2 + 4 + 8 = 14 seconds, which sounds reasonable for normal use cases.